### PR TITLE
Fix `thenReturn(())`, `doNothing()` not returning a nil value

### LINKED ISSUE #32903: [Testerina] Function mocking `thenReturn` isnt compatible with `()` nil return value
**Description:**
Having `thenReturn(())` does not return a nil value but rather causes the inbuilt validation to fail as it considers this as "return value not being set". 

Same applies to `doNothing()` function which is supposed to skip execution entirely.

**Steps to reproduce:**

**Affected Versions:**
`beta2`

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by gayaldassanayake**
## Purpose
$subject

Fixes #32903

## Approach
Removed the validation that checks if returnVal == null and throws an error.

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/FunctionMock.java [hunk 1]
```diff
@@ -73,15 +73,6 @@ public static Object mockHandler(BObject mockFuncObj, Object... args) {
                 break;
             }
         }
-        if (returnVal == null) {
```
Comment (Dilhasha) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/FunctionMock.java:76]: Can we still detect if there is no return value registered with this change?

i.e. when the mock function is defined but `thenReturn`/`doNothing` is not specified within a test case?

Comment (gayaldassanayake) [reply misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/FunctionMock.java:76]: It cannot differentiate between the two scenarios. Will work on a fix for that 👍

Comment (gayaldassanayake) [reply misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/FunctionMock.java:76]: Updated the logic to differentiate two scenarios and updated tests to match each case.

**Review by Dilhasha — APPROVED** [review]

**Comment by gayaldassanayake** [comment]
To get the Windows checks passing, need to rebase once https://github.com/ballerina-platform/ballerina-lang/pull/39665 and https://github.com/ballerina-platform/ballerina-lang/pull/39765 are merged.

**Review by azinneera — DISMISSED** [review]

**Review by azinneera — APPROVED** [review]

**Pull request merged**